documentcloud/addons/models.py

```diff
@@ -267,46 +267,50 @@ def find_run_id(self):
             "%Y-%m-%dT%H:%M"
         )
 
-        page = 1
-        resp = requests.get(
-            f"{self.addon.api_url}/actions/runs?created=%3E{date_filter}",
-            headers=self.addon.api_headers,
-        )
-        resp.raise_for_status()
-        runs = resp.json()["workflow_runs"]
+        url = f"{self.addon.api_url}/actions/runs?created=%3E{date_filter}"
+        while url is not None:
+            logger.info("[FIND RUN ID] get %s", url)
+            resp = requests.get(url, headers=self.addon.api_headers)
+            resp.raise_for_status()
+            url = resp.links.get("next", {}).get("url")
+            resp_json = resp.json()
+            runs = resp_json["workflow_runs"]
+            logger.info("[FIND RUN ID] total_count %s", resp_json["total_count"])
 
-        logger.info("[FIND RUN ID] len(runs) %s", len(runs))
-        while len(runs) > 0:
             for workflow in runs:
                 jobs_url = workflow["jobs_url"]
                 logger.info("[FIND RUN ID] get jobs_url %s", jobs_url)
 
+                cache_key = f"find_run_id:{jobs_url}"
```

Comment:
If the JSON for `jobs_url` is `null`, it looks like around line 311, a cache entry will be added for `find_run_id:None`. Then, later if there was another `jobs_url` that was `null`, this would look up and use this value.

I'm bringing it up just in case, but I'm guessing that's unlikely to happen.